bin/configure-tower-projects.py

```diff
@@ -640,15 +679,101 @@ def add_member(self, user: str) -> dict:
         self.members[user] = member
         return member
 
+    def add_member_to_team(self, team_id: int, user: str) -> int:
+        """Add user to given team within an organization
+
+        Args:
+            team_id (int): Team identifier
+            user (str): Email address for the user
+
+        Returns:
+            int: Team member identifier, which is the same as the
+                organization member identifier
+        """
+        endpoint = f"/orgs/{self.id}/teams/{team_id}/members"
+        data = {"userNameOrEmail": user}
+        response = self.tower.request("POST", endpoint, json=data)
+        # If the user is already a member, you get the following message:
+        #   "The member is already associated with the team"
+        # If this happens, just retrieve the member ID from the organization
+        if "message" in response and "already" in response["message"]:
+            member = self.add_member(user)
+            member_id = member["memberId"]
+        else:
+            member_id = response["member"]["memberId"]
+        return member_id
+
+    def remove_member_from_team(self, team_id: int, member_id: int) -> Dict:
+        """Remove a member from an organization team
+
+        Args:
+            team_id (int): Team identifier
+            member_id (int): Member identifier
+        """
+        endpoint = f"/orgs/{self.id}/teams/{team_id}/members/{member_id}/delete"
+        response = self.tower.request("DELETE", endpoint)
+        return response
+
+    def create_team(self, team_name: str) -> int:
+        """Create team under organization with the given name
+
+        Args:
+            team_name (str): Team name
+
+        Returns:
+            int: Team identifier
+        """
+        # Check if the team already exists
+        endpoint = f"/orgs/{self.id}/teams"
```

Comment:
Nit: I saw this `/orgs/{orgId}/teams/validate`: Check that is a valid team name. 

Does this mean check if team exists? Or if the actual team name is compliant with nextflow tower.

Reply:
It does alert you when the team name already exists, and it also checks if the team name contains valid characters. The former case is handled with the existing code, and the latter case shouldn't happen because we're appending `-<role>` to the stack name, which already only consists of valid characters. 

I could use the dedicated `/validate` endpoint to check if a team name already exists. Can you think of any benefits? 

```
>>> tower.request("GET", "/orgs/193738885884191/teams/validate?name=example-dev-project-maintainers")
{'message': "A team with name 'example-dev-project-maintainers' already exists"}

>>> tower.request("GET", "/orgs/193738885884191/teams/validate?name=#winning")
{'message': 'Team name must contain a minimum of 1 and a maximum of 38 alphanumeric characters separated by dashes or underscores'}
```

Reply:
Are the teams scoped to the tower project? The efficiency would be not having to loop through a list of teams as the team list grows but it's really not too big of a burden

Reply:
> Are the teams scoped to the tower project?

There is one team per project per role (_i.e._ read/write aka maintainer and read-only aka viewer). These teams exist under the organization and aren't tied to the project workspace until they're added to the workspace as participants with the relevant role. I'm not sure if this answers your question. 

We'll need to iterate through the teams anyways to make sure their membership is up-to-date. We can see what the runtime is once we deploy to production. The dev environment only has two teams, so it's hard to tell.